turbo/src/main/kotlin/dev/hotwire/turbo/util/TurboUriHelper.kt

```diff
@@ -11,15 +11,19 @@ import java.io.IOException
 
 internal class TurboUriHelper(val context: Context) {
     @Suppress("BlockingMethodInNonBlockingContext") // https://youtrack.jetbrains.com/issue/KT-39684
-    suspend fun writeFileTo(uri: Uri, directory: File): File? {
+    suspend fun writeFileTo(uri: Uri, destDirectory: File): File? {
         val uriAttributes = getAttributes(uri) ?: return null
 
         return withContext(dispatcherProvider.io) {
-            val file = File(directory, uriAttributes.fileName).also {
-                if (it.exists()) it.delete()
-            }
+            val file = File(destDirectory, uriAttributes.fileName)
```

Comment:
Maybe move this line into the `try` block so the logic is all contained there now?